tests/unit/test_instance_loader.py

```diff
@@ -79,13 +79,50 @@ def test_instanceloader_yaml_data(tmp_path, filename, default_filetype, open_wid
     ],
 )
 def test_instanceloader_toml_data(tmp_path, filename, default_filetype, open_wide):
-    f = tmp_path / "foo.toml"
+    f = tmp_path / filename
     f.write_text('[foo]\nbar = "baz"\n')
     loader = InstanceLoader(open_wide(f), default_filetype=default_filetype)
     data = list(loader.iter_files())
     assert data == [(str(f), {"foo": {"bar": "baz"}})]
 
 
+@pytest.mark.parametrize(
+    "filename, force_filetype",
+    [
+        ("foo.test", "toml"),
+        ("foo", "toml"),
+    ],
+)
+def test_instanceloader_force_filetype_toml(
+    tmp_path, filename, force_filetype, open_wide
+):
+    f = tmp_path / filename
+    f.write_text('[foo]\nbar = "baz"\n')
+    loader = InstanceLoader(open_wide(f), force_filetype=force_filetype)
+    data = list(loader.iter_files())
+    assert data == [(str(f), {"foo": {"bar": "baz"}})]
+
+
+@pytest.mark.parametrize(
+    "filename, force_filetype",
+    [
+        ("foo.test", "json5"),
+        ("foo.json", "json5"),
+    ],
+)
+def test_instanceloader_force_filetype_json(
+    tmp_path, filename, force_filetype, open_wide
+):
+    if not JSON5_ENABLED:
+        pytest.skip("test requires json5")
```

Comment:
There's a slightly nicer way of writing a conditional skip in `pytest`: `pytest` provides a `skipif` decorator which evaluates a condition and marks the test as skipped if true.
e.g.,
https://github.com/python-jsonschema/check-jsonschema/blob/031cc9ad40bd4d873bf2a32ec9d5e454653651d7/tests/acceptance/test_nonjson_schema_handling.py#L54